src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -90,68 +88,45 @@ const StudentApplication = ({ userProfile, authentication }) => {
-            setApplicationID(newApplicationID);
+    const loadApplication = async () => {
+      try {
+        setLoading(true);
+        // Check if the current user is on an application. Returns the application ID number if found
+        const newApplicationID = await housing.getCurrentApplicationID();
+        if (newApplicationID === null || newApplicationID === -1) {
+          // Intentionally trigger the 'catch'
+          throw new Error("Invalid value of 'newApplicationID' = " + newApplicationID);
+        } else {
+          setApplicationID(newApplicationID);
+          setUnsavedChanges(false);
+          const newApplicationDetails = await housing.getApartmentApplication(newApplicationID);
+          if (newApplicationDetails) {
+            setApplicationDetails(newApplicationDetails);
+            setDateSubmitted(newApplicationDetails.DateSubmitted ?? null);
+            setDateModified(newApplicationDetails.DateModified ?? null);
+            setEditorUsername(newApplicationDetails.EditorUsername ?? null);
+            setApplicants(newApplicationDetails?.Applicants ?? []);
+            setPreferredHalls(newApplicationDetails?.ApartmentChoices ?? []);
             setUnsavedChanges(false);
-            housing
-              .getApartmentApplication(newApplicationID)
-              .then((newApplicationDetails) => {
-                if (isSubscribed && newApplicationDetails) {
-                  setApplicationDetails(newApplicationDetails);
-                  setDateSubmitted(newApplicationDetails.DateSubmitted ?? null);
-                  setDateModified(newApplicationDetails.DateModified ?? null);
-                  setEditorUsername(newApplicationDetails.EditorUsername ?? null);
-                  setApplicants((prevApplicants) => {
-                    if (newApplicationDetails.Applicants?.length > 0) {
-                      return newApplicationDetails.Applicants.map(async (newApplicantInfo) => {
-                        if (newApplicantInfo.Profile !== null) {
-                          return newApplicantInfo;
-                        } else {
-                          const newApplicantProfile = await user.getProfileInfo(
-                            newApplicantInfo.Username,
-                          );
-                          return { Profile: newApplicantProfile, ...newApplicantInfo };
-                        }
-                      });
-                    } else {
-                      return prevApplicants;
-                    }
-                  });
-                  setPreferredHalls(newApplicationDetails?.ApartmentChoices ?? []);
-                  setUnsavedChanges(false);
-                }
-              })
-              .catch((error) => {
-                handleError(error);
-              });
           }
-        })
-        .catch((error) => {
-          handleError(error);
-        });
-      setLoading(false);
-    }
+        }
+      } catch (error) {
+        handleError(error);
```

Comment:
~~Since this now only needs to be done in one place, is there any reason to keep it as separate helper function, rather than just putting the contents inside this `catch { ... }` directly?~~
I now this it would be better to change the name of this function, and have it called directly inside the `if (newApplicationID === null)` rather than attempting to trigger it by manually throwing an error.
The new name also helps clarify the true purpose of this function

Reply:
```suggestion
        initializeNewApplication();
```